test/extensions/access_loggers/file/config_test.cc

```diff
@@ -73,6 +74,86 @@ TEST(FileAccessLogConfigTest, FileAccessLogTest) {
   EXPECT_NE(nullptr, dynamic_cast<FileAccessLog*>(instance.get()));
 }
 
+TEST(FileAccessLogConfigTest, FileAccessLogJsonTest) {
+  envoy::config::filter::accesslog::v2::AccessLog config;
+
+  envoy::config::accesslog::v2::FileAccessLog fal_config;
+  fal_config.set_path("/dev/null");
+
+  ProtobufWkt::Value string_value;
+  string_value.set_string_value("%PROTOCOL%");
+
+  auto json_format = fal_config.mutable_json_format();
+  (*json_format->mutable_fields())[std::string{"protocol"}] = string_value;
+
+  EXPECT_EQ(fal_config.access_log_format_case(),
+            envoy::config::accesslog::v2::FileAccessLog::kJsonFormat);
+  MessageUtil::jsonConvert(fal_config, *config.mutable_config());
+
+  NiceMock<Server::Configuration::MockFactoryContext> context;
+  EXPECT_THROW_WITH_MESSAGE(AccessLog::AccessLogFactory::fromProto(config, context), EnvoyException,
+                            "Provided name for static registration lookup was empty.");
+
+  config.set_name(AccessLogNames::get().File);
+
+  AccessLog::InstanceSharedPtr log = AccessLog::AccessLogFactory::fromProto(config, context);
+
+  EXPECT_NE(nullptr, log);
+  EXPECT_NE(nullptr, dynamic_cast<FileAccessLog*>(log.get()));
+
+  config.set_name("INVALID");
+
+  EXPECT_THROW_WITH_MESSAGE(AccessLog::AccessLogFactory::fromProto(config, context), EnvoyException,
+                            "Didn't find a registered implementation for name: 'INVALID'");
+}
+
+TEST(FileAccessLogConfigTest, FileAccessLogJsonConversionTest) {
+  {
+    // Make sure we fail if you set a bool value in the format dictionary
+    envoy::config::filter::accesslog::v2::AccessLog config;
+    config.set_name(AccessLogNames::get().File);
+    envoy::config::accesslog::v2::FileAccessLog fal_config;
+    fal_config.set_path("/dev/null");
+
+    ProtobufWkt::Value bool_value;
+    bool_value.set_bool_value(false);
+    auto json_format = fal_config.mutable_json_format();
+    (*json_format->mutable_fields())[std::string{"protocol"}] = bool_value;
+
+    MessageUtil::jsonConvert(fal_config, *config.mutable_config());
+    NiceMock<Server::Configuration::MockFactoryContext> context;
+
+    EXPECT_THROW_WITH_MESSAGE(AccessLog::AccessLogFactory::fromProto(config, context),
+                              EnvoyException,
+                              "Only string values are supported in the JSON access log format.");
+  }
+
+  {
+    // Make sure we fail if you set a nested Struct value in the format dictionary
+    envoy::config::filter::accesslog::v2::AccessLog config;
+    config.set_name(AccessLogNames::get().File);
+    envoy::config::accesslog::v2::FileAccessLog fal_config;
+    fal_config.set_path("/dev/null");
+
+    ProtobufWkt::Value string_value;
+    string_value.set_string_value(std::string{"some_nested_value"});
+
+    ProtobufWkt::Value struct_value;
+    (*struct_value.mutable_struct_value()->mutable_fields())[std::string{"some_nested_key"}] =
```

Comment:
Nit: do you need these `std::string` conversions?

Reply:
Good catch! I must have been doing something else wrong earlier when I put them in. Removed